src/data/actions/loginForm.js

```diff
@@ -0,0 +1,64 @@
+import axios from 'axios';
+import qs from 'qs';
+
+import {
+  SET_LOGIN_ERROR,
+  SET_LOGIN_PENDING,
+  SET_LOGIN_SUCCESS,
+} from '../constants/loginForm';
+
+const setLoginPending = isLoginPending => ({
+  type: SET_LOGIN_PENDING,
+  isLoginPending,
+});
+
+const setLoginSuccess = isLoginSuccess => ({
+  type: SET_LOGIN_SUCCESS,
+  isLoginSuccess,
+});
+
+const setLoginError = loginError => ({
+  type: SET_LOGIN_ERROR,
+  loginError,
+});
+
+const authenticate = (email, passphrase, callback) => {
+  const dataToSend = {
+    grant_type: 'password',
+    username: email,
+    password: passphrase,
+    client_id: 'N5BfXPPeCNUeheyVrN0sl0VE5MpLD3XlwQ4XrXBW',
+    token_type: 'jwt',
+  };
+  setTimeout(() => {
+    const authUrl = 'https://business.sandbox.edx.org/oauth2/access_token';
+    axios.post(authUrl, qs.stringify(dataToSend))
+      .then((response) => {
+        localStorage.setItem('access_token', response.data.access_token);
+        return callback(null);
+      })
+      .catch(error => (
+        callback(new Error(error))
+      ));
+  }, 2000);
+};
+
+const login = (email, password) => (
+  (dispatch) => {
+    dispatch(setLoginPending(true));
+    // follow authentication process for this
+    authenticate(email, password, (error) => {
```

Comment:
Not really necessary to use a separate method `authenticate` with a callback here to make this work. Refer to how the [`fetchCourseEnrollments` ](https://github.com/edx/edx-portal/blob/master/src/data/actions/courseEnrollments.js#L18) action does this.

Reply:
@jonalexvasquez ^^ Let's move the `authenticate` method to the API services layer like `fetchCourseEnrollments` does (see above link).